Source/objects.cpp

```diff
@@ -3821,7 +3823,7 @@ void InitObjects()
 	} else {
 		ApplyObjectLighting = true;
 		AdvanceRndSeed();
-		if (currlevel == 9 && !gbIsMultiplayer)
+		if (currlevel == 9 && !UseMultiplayerQuests())
 			AddSlainHero();

```

Comment:
can move the `AdvanceRndSeed()` call here to an else statement to end up with the same number of rng calls